grapple/types/pages.py

```diff
@@ -169,6 +169,66 @@ class Meta:
         interfaces = (PageInterface,)
 
 
+def get_preview_page(token):
+    """
+    Get a preview page from a token.
+    """
+    token_params_list = token.split(":").pop(0)
+    token_params_kvstr = token_params_list.split(";")
+
+    params = {}
+    for arg in token_params_kvstr:
+        key, value = arg.split("=")
+        params[key] = value
+
+    """
+    Is there a more convenient way to get at the get_page_from_preview_token
+    method? 
+    """
+
+    """ 
+    We get id when previewing a page that has been saved. We're going to use it
+    to get the page so we can lookup it's class in order to call 
+    get_page_from_preview_token
+    """
+    id = params.get("id")
+    if id:
+        page = WagtailPage.objects.get(pk=id).specific
```

Comment:
please wrap this and the ctype one in a try/except `<model>.DoesNotExist`

Reply:
I just wrapped the whole thing in a try/except.